cdap-ui/app/cdap/components/FieldLevelLineage/v2/Context/FllContextHelper.ts

```diff
@@ -367,3 +414,41 @@ export function getOperations(
     cb(operations);
   });
 }
+
+export function fetchUnrelatedFields(
+  namespace: string,
+  tablename: string,
+  type: string,
+  relatedFields: IField[],
+  start: ITimeType,
+  end: ITimeType,
+  cb: (fields: IField[]) => void
+) {
+  // Filter out fields that are already being rendered to maintain field order
+  const getUnrelatedFields = (allFields: string[]) => {
+    const isRelatedField = (fieldId, fields) => {
+      return fields.filter((field) => field.id === fieldId).length > 0;
+    };
+
+    const unrelatedFields = allFields.filter((fieldname) => {
+      const fieldId = getFieldId(fieldname, tablename, namespace, type);
+      return !isRelatedField(fieldId, relatedFields);
+    });
+    return unrelatedFields.map((fieldname) => {
+      const fieldProperties: IField = {
+        type,
+        namespace,
+        name: fieldname,
+        dataset: tablename,
+        id: getFieldId(fieldname, tablename, namespace, type),
+      };
+      return fieldProperties;
+    });
+  };
+  const params = { namespace, entityId: tablename, start, end };
+
+  MyMetadataApi.getFields(params).subscribe((res) => {
+    const unrelatedFields = getUnrelatedFields(res);
+    cb(unrelatedFields);
```

Comment:
rather than having a callback here, it's cleaner to have this function simply return the observable and let whoever is calling it to subscribe to the actual response. When you do this, you can't handle errors in the API. 

I see that you are transforming the response, so you can use map operator to the response of the observable before returning it

Reply:
I've made the change to use an observable instead of a callback. There isn't any real error handling yet, though.